plugins/secretstores/gdchauth/gdchauth_test.go

```diff
@@ -0,0 +1,314 @@
+		TokenURI:            tokenURI,
+		Project:             testProject,
+	}
+
+	keyData, err := json.Marshal(saKey)
+	require.NoError(t, err)
+
+	tmpfile, err := os.CreateTemp(t.TempDir(), "test-sa-key-*.json")
+	require.NoError(t, err)
+	defer tmpfile.Close()
+
+	_, err = tmpfile.Write(keyData)
+	require.NoError(t, err)
+
+	return tmpfile.Name()
+}
+
+// --- Test Cases ---
+
+func TestGdchAuth_Init(t *testing.T) {
+	t.Run("missing service account file should fail", func(t *testing.T) {
+		g := &GdchAuth{Log: testutil.Logger{}}
+		err := g.Init()
+		require.Error(t, err)
+		require.EqualError(t, err, "service_account_file is required")
+	})
+
+	t.Run("non-existent service account file should fail", func(t *testing.T) {
+		g := &GdchAuth{
+			ServiceAccountFile: "non-existent-file.json",
+			Log:                testutil.Logger{},
+		}
+		err := g.Init()
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "failed to read service account file")
+	})
+
+	t.Run("invalid service account file json should fail", func(t *testing.T) {
+		tmpfile, err := os.CreateTemp(t.TempDir(), "invalid-sa-key-*.json")
+		require.NoError(t, err)
+		defer os.Remove(tmpfile.Name())
+		_, err = tmpfile.WriteString("this is not json")
+		require.NoError(t, err)
+		require.NoError(t, tmpfile.Close())
+
+		g := &GdchAuth{
+			ServiceAccountFile: tmpfile.Name(),
+			Log:                testutil.Logger{},
+		}
+		err = g.Init()
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "failed to parse service account JSON")
+	})
+
+	t.Run("invalid private key pem should fail", func(t *testing.T) {
+		saKey := ServiceAccountKey{
+			PrivateKey: "this is not a pem",
+		}
+		keyData, err := json.Marshal(saKey)
+		require.NoError(t, err)
+
+		tmpfile, err := os.CreateTemp(t.TempDir(), "test-sa-key-*.json")
+		require.NoError(t, err)
+		defer os.Remove(tmpfile.Name())
+		_, err = tmpfile.Write(keyData)
+		require.NoError(t, err)
+		require.NoError(t, tmpfile.Close())
+
+		g := &GdchAuth{
+			ServiceAccountFile: tmpfile.Name(),
+			Log:                testutil.Logger{},
+		}
+		err = g.Init()
+		require.Error(t, err)
+		require.EqualError(t, err, "failed to decode PEM block from private key")
+	})
+
+	t.Run("successful init should set defaults", func(t *testing.T) {
+		keyFile := generateTestKeyFile(t, "http://localhost/token")
+		defer os.Remove(keyFile)
+
+		g := &GdchAuth{
+			ServiceAccountFile: keyFile,
+			Log:                testutil.Logger{},
+		}
+		err := g.Init()
+		require.NoError(t, err)
+		require.NotNil(t, g.saKey)
+		require.NotNil(t, g.httpClient)
+		require.Equal(t, config.Duration(5*time.Minute), g.TokenExpiryBuffer)
+	})
+}
+
+func TestGdchAuth_GetToken(t *testing.T) {
+	// --- Setup Mock Token Server ---
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Check request method and content type
+		if !assert.Equal(t, "POST", r.Method) {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
```

Comment:
No `assert` please, we only use `require` which cannot be used here. So please do


```suggestion
		if "POST" != r.Method {
			t.Fail()
			w.WriteHeader(http.StatusBadRequest)
			return
		}
```

Same for other instances below!